dns/jsonify/main.go

```diff
@@ -0,0 +1,242 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/golang/glog"
+	"gopkg.in/yaml.v2"
+	"k8s.io/kubernetes/test/e2e/perftype"
+)
+
+const (
+	// S_TO_MS is a second to milisecond ratio.
+	S_TO_MS = 1000
+)
+
+// BenchamrkResult is a dns benchmark results structure.
+type BenchamrkResult struct {
+	Code   int             `yaml:"code"`
+	Data   BenchmarkData   `yaml:"data"`
+	Params BenchmarkParams `yaml:"params"`
+}
+
+// BenchmarkData represents dns benchmark data.
+type BenchmarkData struct {
+	Latency50Percentile float64 `yaml:"latency_50_percentile"`
+	Latency95Percentile float64 `yaml:"latency_95_percentile"`
+	Latency99Percentile float64 `yaml:"latency_99_percentile"`
+	AvgLatency          float64 `yaml:"avg_latency"`
+	MaxLatency          float64 `yaml:"max_latency"`
+	MinLatency          float64 `yaml:"min_latency"`
+	Qps                 float64 `yaml:"qps"`
+	QueriesCompleted    float64 `yaml:"queries_completed"`
+	QueriesLost         float64 `yaml:"queries_lost"`
+	QueriesSent         float64 `yaml:"queries_sent"`
+}
+
+// BenchmarkParams represents dns benchmark params.
+type BenchmarkParams struct {
+	RunLengthSeconds float64  `yaml:"run_length_seconds"`
+	QueryFile        string   `yaml:"query_file"`
+	KubednsCpu       *float64 `yaml:"kubedns_cpu"`
+	DnsmasqCpu       *float64 `yaml:"dnsmasq_cpu"`
+	DnsmasqCache     *float64 `yaml:"dnsmasq_cache"`
+	MaxQps           *float64 `yaml:"max_qps"`
+}
+
+func main() {
+	defer glog.Flush()
+	err := run()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func run() error {
+	var benchmarkDirPath string
+	var jsonDirPath string
+	var benchmarkName string
+	flag.CommandLine = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	flag.StringVar(&benchmarkDirPath, "benchmarkDirPath", ".", "benchmark results directory path")
+	flag.StringVar(&jsonDirPath, "jsonDirPath", ".", "json results directory path")
+	flag.StringVar(&benchmarkName, "benchmarkName", ".", "benchmark name")
+
+	if err := flag.CommandLine.Parse(os.Args[1:]); err != nil {
+		return fmt.Errorf("flag parse failed: %v", err)
+	}
+
+	glog.Infof("benchmarkDirPath: %v\n", benchmarkDirPath)
+	glog.Infof("jsonDirPath: %v\n", jsonDirPath)
+	glog.Infof("benchmarkName: %v\n", benchmarkName)
+
+	latency := perftype.PerfData{Version: "v1"}
+	latencyPerc := perftype.PerfData{Version: "v1"}
+	queries := perftype.PerfData{Version: "v1"}
+	qps := perftype.PerfData{Version: "v1"}
+
+	fileList, err := getFileList(benchmarkDirPath)
+	if err != nil {
+		return fmt.Errorf("listing files error: %v", err)
+	}
+
+	for _, file := range fileList {
+		glog.Infof("processing %v\n", file)
```

Comment:
Use %s to print since we know "file" is a string

Reply:
Done.